### ISSUE TITLE
salt 2018.3.0 inverts diff with test=True

### ISSUE THREAD

**Issue by kt97679** [saltstack/salt#47551]
### Description of Issue/Question

Here is very simple salt state:
```
file-example:
  file.managed:
    - name: /var/tmp/test-file.txt
    - source: salt://salt-demo/test-file.txt
```
with test-file.txt having following content:
```
this is a test file
```
I apply this salt state and get correct file on the target. Now I modify test-file.txt:
```
111this is a test file
```
and run preview with test=True. I'm getting this:
```
iad-kvt-dev-01:
----------
          ID: file-example
    Function: file.managed
        Name: /var/tmp/test-file.txt
      Result: None
     Comment: The file /var/tmp/test-file.txt is set to be changed
     Started: 15:33:30.562177
    Duration: 161.73 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1 +1 @@
                  -111this is a test file
                  +this is a test file
----------
          ID: salt-demo.file::goal
    Function: stateconf.set
      Result: True
     Comment: 
     Started: 15:33:30.725167
    Duration: 0.849 ms
     Changes:   

Summary for iad-kvt-dev-01
------------
Succeeded: 2 (unchanged=1, changed=1)
Failed:    0
------------
Total states run:     2
Total run time: 162.579 ms
```
As you can see salt inverts data in the output. Old data is shown as new and new as old. Now I apply this salt state and I see this:
```
iad-kvt-dev-01:
----------
          ID: file-example
    Function: file.managed
        Name: /var/tmp/test-file.txt
      Result: True
     Comment: File /var/tmp/test-file.txt updated
     Started: 15:34:16.385187
    Duration: 202.136 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1 +1 @@
                  -this is a test file
                  +111this is a test file
----------
          ID: salt-demo.file::goal
    Function: stateconf.set
      Result: True
     Comment: 
     Started: 15:34:16.590462
    Duration: 1.537 ms
     Changes:   

Summary for iad-kvt-dev-01
------------
Succeeded: 2 (changed=1)
Failed:    0
------------
Total states run:     2
Total run time: 203.673 ms
```
Here change is shown correctly and file on the target is in the correct state.

### Setup
Provided above.

### Steps to Reproduce Issue
Shown above.

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

Please note, that I use salt 2018.3.0 with couple of minor patches on top of it (like adding logging to the file on the target), here is version report:

Salt Version:
           Salt: 2018.3.0-6-g155769a
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15 (default, May  2 2018, 09:46:29)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6
 
System Versions:
           dist: debian jessie/sid 
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-119-generic
         system: Linux
        version: debian jessie/sid

**Comment by tbaker57** [comment]
Is this a duplicate of #46931 (and #47385, #47016) ? If so, looks like it will be fixed in 2018.3.1

cheers
Tim

**Comment by kt97679** [comment]
Oh, sorry, I tried to find if this was filed but missed those. Closing now.
